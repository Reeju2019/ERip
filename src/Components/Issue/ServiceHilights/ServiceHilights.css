.divContainer{
  width: 35vw;
}

.itemImg{
  width: 4vmax;
}

.itemPara{
  font-size: 12px;
  color: gray;
  margin: 0;
}