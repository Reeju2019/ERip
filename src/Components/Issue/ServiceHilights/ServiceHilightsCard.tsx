import * as React from 'react'

interface IServiceHilightsCardProps {
  item: {
    title: string
    text: string
    image: string
  }
}

const ServiceHilightsCard: React.FunctionComponent<IServiceHilightsCardProps> = ({ item }) => {
  return (
    <>
      <div className='d-flex flex-row divContainer p-2 my-4'>
        <div>
          <img src={item.image} className='itemImg' alt='itemImg' />
        </div>
        <div className='p-2'>
          <h6>{item.title}</h6>
          <p className='itemPara'>{item.text}</p>
        </div>
      </div>
    </>
  )
}

export default ServiceHilightsCard
