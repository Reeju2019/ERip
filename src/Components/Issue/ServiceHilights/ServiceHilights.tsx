import { Container, Row } from 'react-bootstrap'
import './ServiceHilights.css'
import IssueData from '../../../Data/IssuePage.mock.json'
import ServiceHilightsCard from './ServiceHilightsCard'

const ServiceHilights: React.FunctionComponent = () => {
  return (
    <>
      <section className='p-5'>
        <Container>
          <Row>
            <h4 className='d-flex justify-content-center mt-3'>Service Highlights</h4>
          </Row>
          <Row>
            <div className='d-flex  flex-wrap justify-content-center'>
              {IssueData.constData.serviceHilights.map((item, index) => {
                return (
                  <div key={index}>
                    <ServiceHilightsCard item={item} />
                  </div>
                )
              })}
            </div>
          </Row>
        </Container>
      </section>
    </>
  )
}

export default ServiceHilights
