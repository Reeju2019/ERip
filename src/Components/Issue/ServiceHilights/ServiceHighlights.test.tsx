import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ServiceHilights from './ServiceHilights'


test('service highlights', () => {
    render(<ServiceHilights />)
})
describe('service highlights rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<ServiceHilights />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing service highlights', () => {
        const div = document.createElement('div')
        render(<ServiceHilights />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
