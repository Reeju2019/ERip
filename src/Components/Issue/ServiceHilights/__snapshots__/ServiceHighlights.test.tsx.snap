// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service highlights rendering specification SnapShot testing 1`] = `
<section
  className="p-5"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h4
        className="d-flex justify-content-center mt-3"
      >
        Service Highlights
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="d-flex  flex-wrap justify-content-center"
      >
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row1.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                Relentless Support
              </h6>
              <p
                className="itemPara"
              >
                Our professional service providers work round-the-clock to offer you unmatched repair services.
              </p>
            </div>
          </div>
        </div>
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row2.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                Post-Service Payment
              </h6>
              <p
                className="itemPara"
              >
                We allow zero deceptions. Pay for the service after you have received it.
              </p>
            </div>
          </div>
        </div>
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row1.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                Data Privacy
              </h6>
              <p
                className="itemPara"
              >
                We offer a seamless repair service with 100% privacy maintenance.
              </p>
            </div>
          </div>
        </div>
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row2.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                Uncompromised Quality
              </h6>
              <p
                className="itemPara"
              >
                Experience speedy service with uncompromised quality on repair and spare parts.
              </p>
            </div>
          </div>
        </div>
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row1.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                Single Day Service
              </h6>
              <p
                className="itemPara"
              >
                Book a service appointment and receive delivery on the same day.
              </p>
            </div>
          </div>
        </div>
        <div>
          <div
            className="d-flex flex-row divContainer p-2 my-4"
          >
            <div>
              <img
                alt="itemImg"
                className="itemImg"
                src="/assets/whyus/row2.png"
              />
            </div>
            <div
              className="p-2"
            >
              <h6>
                ERIP-certified spare parts
              </h6>
              <p
                className="itemPara"
              >
                When it comes to maintaining the health of your product, there can never be a substitute for genuine parts. ERIP-certified spare parts meet the criteria of the latest quality standards and maintain the stability of your products for the long run.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
